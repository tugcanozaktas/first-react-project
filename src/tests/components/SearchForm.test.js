import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const searchText = "LefkoÅŸa";
  const setSearchText = () => {};
  const onSubmit = () => {};

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={onSubmit}
      />,
    );
    expect(asFragment).toMatchSnapshot();
  });

  it("has correct button text", () => {
    const { getByText } = render(
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={onSubmit}
      />,
    );
    expect(getByText("Search")).toHaveClass("search-form__button");
  });
});
