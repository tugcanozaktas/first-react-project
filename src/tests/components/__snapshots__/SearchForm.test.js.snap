// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `[Function]`;
